### PR TITLE
PLANNER-2389 Upgrade Quarkus to 1.13

### PR DESCRIPTION
**Pull request by rsynek**
### JIRA
https://issues.redhat.com/browse/PLANNER-2389

### Referenced pull requests
https://github.com/kiegroup/optaplanner/pull/1254
https://github.com/kiegroup/optaplanner-quickstarts/pull/93
https://github.com/kiegroup/optaweb-employee-rostering/pull/602
https://github.com/kiegroup/optaweb-vehicle-routing/pull/471

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,12 +55,6 @@
     <version.npm>6.14.4</version.npm>
     <version.org.apache.poi>4.1.2</version.org.apache.poi>
     <version.org.jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec>1.0.1.Final</version.org.jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec>
-    <version.org.mockito>3.7.0</version.org.mockito>
-    <version.org.hibernate>5.4.28.Final</version.org.hibernate>
```
Comment (Christopher-Chianelli) [inline pom.xml:59]: The hibernate version in optaplanner-bom is incompatible with the hibernate version in Quarkus; the hibernate version here overrides the optaplanner-bom one so it matches the Quarkus version. As such, this property need to stay until it matches.

**Review by Christopher-Chianelli — CHANGES_REQUESTED** [review]
You need to keep the hibernate version; there a version mismatch between optaplanner-bom and quarkus that it fixes

File: pom.xml (hunk 1 above)
Comment (Christopher-Chianelli) [reply pom.xml:59]: (IIRC, is also the case for mockito)

Comment (rsynek) [reply pom.xml:59]: See the linked PR, which actually upgrades both (and a couple of other versions) to the same version Quarkus declares.
https://github.com/kiegroup/optaplanner/pull/1254

Comment (Christopher-Chianelli) [reply pom.xml:59]: Build log from Jenkins build suggests wrong hibernate version is still used; is it not taking that PR into consideration?

Comment (rsynek) [reply pom.xml:59]: Unfortunately yes; the only CI capable of taking PRs to the other repositories into consideration is the Build Chain, which apparently has its own issues.

**Review by Christopher-Chianelli — APPROVED** [review]
Tested manually; merge with other PR's

**Pull request merged**